src/WebJobs.Script/Binding/GeneralScriptBindingProvider.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.ObjectModel;
+using System.IO;
+using System.Reflection;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Azure.WebJobs.Script.Description;
+using Microsoft.Azure.WebJobs.Script.Extensibility;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Script.Binding
+{
+    // General binder that works directly off SDK interfaces. 
+    class GeneralScriptBindingProvider : ScriptBindingProvider
+    {
+        public ITooling Tooling { get; set; }
```

Comment:
Rather than ITooling, I think we should generalize this as IJobHostMetadataProvider, since it really is about providing metadata for all registered extensions in a JobHost.

Also, rather than an interface, I think we should make this an abstract class so we can add new members w/o breaking changes. I.e. JobHostMetadataProvider